scheduler/src/cook/postgres.clj

```diff
@@ -0,0 +1,98 @@
+(ns cook.postgres
+  (:require [clojure.tools.logging :as log]
+            [clojure.java.jdbc :as sql]
+            [cook.config :as config])
+  (:import (java.lang Runtime)
+           (java.util Random)))
+
+(defn setup-database
+  "Setup a fresh cook schema from scratch. At present, calls out
+  to a bash script, passing it the target schema name and hopes it does the right thing."
+  [setup-script schema-name]
+  ; This environmental variable contains the path of a setup script that, given a schema name, creates a database.
+  (let [^Process p (-> (Runtime/getRuntime)
+                       (.exec (str setup-script " " schema-name)))
+        _ (log/info "Waiting for database schema production process" (.pid p) "to exit")
+        exit-code (.waitFor p)]
+    (assert (= exit-code 0) (str "Exit code for process creation is" exit-code "not zero"))))
+
+(defn make-database-connection-dictionary-for-unit-tests
+  "Given a target schema name and assuming a variety of environmental
+   variables are set, creates the metadata to connect to a database."
+  [currentSchema]
+  (merge {:dbtype "postgresql"
+          :dbname (or (System/getenv "COOK_DB_TEST_PG_DB") "cook_local")
+          ; Note that we need to save the currentSchema in the connection info to
+          :currentSchema currentSchema
+          :host (or (System/getenv "COOK_DB_TEST_PG_SERVER") "127.0.0.1")
+          :ssl false
+          :sslfactory "org.postgresql.ssl.NonValidatingFactory"}
+         ; PGPASSWORD is the default enviornmental variable used by postgres psql.
+         (when-let [passwd (or  (System/getenv "COOK_DB_TEST_PG_PASSWORD")
+                                (System/getenv "PGPASSWORD"))]
+           {:password passwd})
+         (when-let [username (System/getenv "COOK_DB_TEST_PG_USER")]
+           {:user username})))
+
+
+; With unit tests, we need to create a new configuration dictionary in the test fixture.
+; That gets stored here.
+(def saved-pg-config-dictionary (atom nil))
```

Comment:
Shouldn't this be under `scheduler/test/` instead of `scheduler/src/`?

Reply:
No. (pg-db) is how Cook gets its database connection and has to be here. This is read by that function, and is how the test fixture code communicates the chosen database out.

Reply:
The comment seems to imply this is only for unit tests

Reply:
This code is called by Cook's prod code (via pg-db), so needs to be in the call chain for Cook. This codepath however is only actually invoked when run within a unit test context.